File: uri/1088.cpp
#include <bits/stdc++.h>
#define MAXN (int)1e5 + 1

using namespace std;

int ans = 0;
string players[] = {"Carlos", "Marcelo"};

int* sliceArray(int arr[], int start, int end) {
  int* slice = new int[end - start];

  for (int i = 0; i < end - start; i++) {
    slice[i] = arr[start + i];
  }

  return slice;
}

void merge(int arr[], int start, int middle, int end) {
  int* left = sliceArray(arr, start, middle);
  int* right = sliceArray(arr, middle, end);

  int i = 0, j = 0;

  for (int index = start; index < end; index++) {
    if (i >= (middle - start)) {
      arr[index] = right[j++];
    } else if (j >= (end - middle)) {
      arr[index] = left[i++];
    } else if (left[i] < right[j]) {
      arr[index] = left[i++];
    } else {
      arr[index] = right[j++];
      ans += middle - start - i;
    }
  }
}

void mergesort(int arr[], int start, int end) {
  int middle = (start + end) / 2;
  if (end - start > 1) {
    mergesort(arr, start, middle);
    mergesort(arr, middle, end);
    merge(arr, start, middle, end);
  }
}

int main() {

  int n, numbers[MAXN];

  while(cin >> n && n > 0) {
    ans = 0;
    for (int i = 0; i < n; i++) {
      cin >> numbers[i];
    }

    mergesort(numbers, 0, n);

    cout << (ans % 2 == 0 ? "Carlos" : "Marcelo") << endl;
  }

  return 0;
}

File: uri/1088.py
def merge(arr, start, middle, end):
  ans = 0
  left = arr[start:middle]
  right = arr[middle:end]

  i = 0
  j = 0

  for k in range(start, end):
    if i >= middle:
      arr[k] = right[j]
      j = j + 1
    elif j >= end:
      arr[k] = left[i]
      i = i + 1
    elif left[i] < right[j]:
      arr[k] = left[i]
      i = i + 1
    else:
      arr[k] = right[j]
      j = j + 1
      ans += len(left) - i
  return ans
  
def mergesort(arr, start, end):
  middle = (start + end) // 2

  if end - start > 1:
    mergesort(arr, start, middle)
    mergesort(arr, middle, end)
    merge(arr, start, middle, end)
  return 0

player = ['Carlos', 'Marcelo']

while True:
  ans = 0
  n = [int(x) for x in raw_input().split(' ')]
  arr = n[1:len(n)]
  if n[0] == 0:
    break
  mergesort(arr, 0, len(arr))
  print player[ans % 2]